driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -611,6 +615,21 @@ void cleanupIdleConnections(long now) {
         }
     }
 
+    @Override
+    protected void checkNotInEventLoop() {
+        if (!CHECK_IO_DEADLOCKS)
+            return;
+        for (EventExecutor executor : cluster.manager.connectionFactory.eventLoopGroup) {
+            if (executor.inEventLoop()) {
+                throw new IllegalStateException(
```

Comment:
Weren't we supposed to log a warning first, then turn it into an exception at the next major release? Maybe you can turn your system property into stg like `IO_DEADLOCK_CHECK_MODE` with values `NONE`, `WARN`, `FAIL`.